core/nlu/model/SnipsNlu.py

```diff
@@ -36,65 +36,67 @@ def convertDialogTemplate(self, file: Path):
+
+			nluTrainingSampleEntity['data'] = [{
+					'value'   : value['value'],
+					'synonyms': value.get('synonyms', list())
+				} for value in entity['values']
+			]
+
+		nluTrainingSample['intents'] = dict()
+		for intent in dialogTemplate['intents']:
+			intentName = f'{self.ConfigManager.getAliceConfigByName("intentsOwner")}:{intent["name"]}'
+			slots = self.loadSlots(intent)
+			nluTrainingSample['intents'].setdefault(intentName, {'utterances': list()})
+
+			for utterance in intent['utterances']:
+				data = list()
+				result = re.findall(self.UTTERANCE_REGEX, utterance)
+				if not result:
+					data.append({
+						'text': utterance
 					})
-				nluTrainingSample['entities'][entity['name']]['data'] = values
-
-			nluTrainingSample['intents'] = dict()
-			for intent in dialogTemplate['intents']:
-				intentName = f'{self.ConfigManager.getAliceConfigByName("intentsOwner")}:{intent["name"]}'
-				slots = self.loadSlots(intent)
-				nluTrainingSample['intents'].setdefault(intentName, dict()).setdefault('utterances', list())
-
-				for utterance in intent['utterances']:
-					data = list()
-					result = re.findall(self.UTTERANCE_REGEX, utterance)
-					if not result:
-						data.append({
-							'text': utterance
-						})
-					else:
-						for dataset in result:
-							for match in dataset:
-								if not match:
-									continue
-
-								if not ':=>' in match:
-									data.append({
-										'text': match
-									})
-								else:
-									slotName = match.split(':=>')[1]
-									entity = slots[slotName] if slotName in slots else 'Unknown'
-
-									if entity.startswith('snips/'):
-										nluTrainingSample['entities'][entity] = dict()
-
-									data.append({
-										'entity'   : entity,
-										'slot_name': slotName,
-										'text'     : match.split(':=>')[0]
-									})
-
-					# noinspection PyTypeChecker
-					nluTrainingSample['intents'][intentName]['utterances'].append({'data': data})
-
-			with Path(self._cachePath, f'{dialogTemplate["skill"]}_{file.stem}.json').open('w') as fpp:
-				fpp.write(json.dumps(nluTrainingSample, indent=4))
+					continue
+				else:
+					for dataset in result:
+						for match in dataset:
+							if not match:
+								continue
+
+							if ':=>' not in match:
+								data.append({
+									'text': match
+								})
+								continue
+							
+							text, slotName = match.split(':=>')
+							entity = slots.get(slotName, 'Unknown')
+
+							if entity.startswith('snips/'):
+								nluTrainingSample['entities'][entity] = dict()
+
+							data.append({
+								'entity'   : entity,
+								'slot_name': slotName,
+								'text'     : text
+							})
+
+				# noinspection PyTypeChecker
+				nluTrainingSample['intents'][intentName]['utterances'].append({'data': data})
+
+			with Path(self._cachePath, f'{dialogTemplate["skill"]}_{file.stem}.json').open('w') as fp:
+				#TODO: this will not allow ascii like Ä in german so this should probably include ensure_ascii=False
```

Comment:
I am pretty sure that this should allow ascii. is this correct @Psychokiller1888

Reply:
That's true, Ascii is needed, good catch